pallets/utility/src/lib.rs

```diff
@@ -871,4 +884,121 @@ impl<T: Config> Pallet<T> {
+        Self::ensure_root(origin)?;
+
+        let as_origin: T::RuntimeOrigin = (*as_origin).into();
+
+        let behalf_account_id = {
+            match as_origin.clone().into() {
+                Ok(RawOrigin::Signed(account_id)) => Some(account_id.clone()),
+                _ => None,
+            }
+        };
+        let behalf_identity = behalf_account_id
+            .clone()
+            .and_then(|acc| Identity::<T>::get_identity(&acc));
+
+        let dispatch_info = call.get_dispatch_info();
+        let call_result = Self::run_with_temporary_did_and_payer(
+            as_origin,
+            behalf_account_id,
+            behalf_identity,
+            call,
+            true,
+        );
+        // Get the actual weight of this call.
+        let weight = extract_actual_weight(&call_result, &dispatch_info);
+
+        Self::deposit_event(Event::<T>::DispatchedAs {
+            result: call_result.map(|_| ()).map_err(|e| e.error),
+        });
+        // POLYMESH: return the actual weight of the call.
+        let base_weight = <T as Config>::WeightInfo::dispatch_as();
+        Ok(Some(base_weight.saturating_add(weight)).into())
+    }
+
+    fn base_as_derivative(
+        origin: T::RuntimeOrigin,
+        index: u16,
+        call: Box<<T as Config>::RuntimeCall>,
+    ) -> DispatchResultWithPostInfo {
+        let origin_account_id = ensure_signed(origin.clone())?;
+
+        // Sets the caller to be the derivative of the caller's account
+        let derivative_account_id = Self::derivative_account_id(origin_account_id, index)?;
+        let mut origin = origin;
+        origin.set_caller_from(frame_system::RawOrigin::Signed(
+            derivative_account_id.clone(),
+        ));
+        // Get the identity of the derivative account id
+        let derivative_did = Identity::<T>::get_identity(&derivative_account_id);
+
+        let dispatch_info = call.get_dispatch_info();
+        let call_result = Self::run_with_temporary_did_and_payer(
+            origin,
+            Some(derivative_account_id),
+            derivative_did,
+            call,
+            false,
+        );
+
+        // Always take into account the base weight of this call and add the real weight of the dispatch.
+        let mut weight = <T as Config>::WeightInfo::as_derivative()
+            .saturating_add(T::DbWeight::get().reads_writes(1, 1));
+        weight = weight.saturating_add(extract_actual_weight(&call_result, &dispatch_info));
+
+        call_result
+            .map_err(|mut err| {
+                err.post_info = Some(weight).into();
+                err
+            })
+            .map(|_| Some(weight).into())
+    }
+
+    /// Derive a derivative account ID from the owner account and the index.
+    pub fn derivative_account_id(
+        origin_account_id: T::AccountId,
+        index: u16,
+    ) -> Result<T::AccountId, DispatchError> {
+        let entropy = (origin_account_id, index).using_encoded(blake2_256);
```

Comment:
```suggestion
        let entropy = (b"modlpy/utilisuba", origin_account_id, index).using_encoded(blake2_256);
```
The prefix is important to prevent the caller from fully controlling the inputs to the hash function.